picard/dropin_sklearn.py

```diff
@@ -0,0 +1,206 @@
+        sources. This is equal to the unmixing matrix when ``whiten`` is
+        False, and equal to ``np.dot(unmixing_matrix, self.whitening_)`` when
+        ``whiten`` is True.
+    mixing_ : ndarray of shape (n_features, n_components)
+        The pseudo-inverse of ``components_``. It is the linear operator
+        that maps independent sources to the data.
+    mean_ : ndarray of shape(n_features,)
+        The mean over features. Only set if `self.whiten` is True.
+    whitening_ : ndarray of shape (n_components, n_features)
+        Only set if whiten is 'True'. This is the pre-whitening matrix
+        that projects data onto the first `n_components` principal components.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_digits
+    >>> from picard import Picard
+    >>> X, _ = load_digits(return_X_y=True)
+    >>> transformer = Picard(n_components=7,
+    ...         random_state=0)
+    >>> X_transformed = transformer.fit_transform(X)
+    >>> X_transformed.shape
+    (1797, 7)
+
+    Notes
+    -----
+    Using a different density than `'tanh'` may lead to erratic behavior of
+    the algorithm: when `extended=True`, the non-linearity used by the
+    algorithm is `x +/- fun(x)`. The non-linearity should correspond to a
+    density, hence `fun` should be dominated by `x ** 2`. Further,
+    `x + fun(x)` should separate super-Gaussian sources and `x-fun(x)`
+    should separate sub-Gaussian sources. This set of requirement is met by
+    `'tanh'`.
+    """
+    def __init__(self, n_components=None, *, ortho=True, extended=None,
+                 whiten=True, fun='tanh', max_iter=200, tol=1e-7,
+                 w_init=None, m=7,  ls_tries=10, lambda_min=0.01,
+                 random_state=None):
+        super().__init__()
+        if max_iter < 1:
+            raise ValueError("max_iter should be greater than 1, got "
+                             "(max_iter={})".format(max_iter))
+        self.n_components = n_components
+        self.ortho = ortho
+        self.extended = extended
+        self.whiten = whiten
+        self.fun = fun
+        self.max_iter = max_iter
+        self.tol = tol
+        self.w_init = w_init
+        self.m = m
+        self.ls_tries = ls_tries
+        self.lambda_min = lambda_min
+        self.random_state = random_state
+
+    def _fit(self, X, compute_sources=False):
+        """Fit the model
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+        compute_sources : bool, default=False
+            If False, sources are not computes but only the rotation matrix.
+            This can save memory when working with big data. Defaults to False.
+
+        Returns
+        -------
+      X_new : ndarray of shape (n_samples, n_components)
```

Comment:
```suggestion
        X_new : ndarray of shape (n_samples, n_components)
```

Reply:
I don't understand, there is a `flake8 picard` in the travis.yml...